datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -303,6 +322,16 @@ fn create_group_accumulator(
     }
 }
 
+/// Extracts a successful Ok(_) or returns Poll::Ready(Some(Err(e))) with errors
+macro_rules! extract_ok {
```

Comment:
One way to avoid this is to extract a sync function that returns a Result, allowing the use of `?` and then map as necessary

Reply:
I tried this - I am not sure it is better -- I will put up a follow on PR https://github.com/apache/arrow-datafusion/pull/7025